krm-functions/interface-fn/fn/testdata/interface_cni_vlan-ipv6only/_expected.yaml

```diff
@@ -32,16 +32,25 @@ items:
       name: vpc-ran
   status: {}
 - apiVersion: kpt.dev/v1
-  info:
-    description: upf package example
   kind: Kptfile
+  # comment A
   metadata:
+    name: pkg-upf
+    #commentB
     annotations:
       config.kubernetes.io/local-config: "true"
-    name: pkg-upf
+  info:
+    description: upf package example
   pipeline: {}
   status:
     conditions:
+    - message: update condition for initial resource
```

Comment:
I am not 100% sure this PR is going to fix the probelm we are seeing. The problem I see is that when the Porch controllers emit the Kptfile, these come out like:

```
- type: workload.nephio.org/v1alpha1.UPFDeployment.upf-cluster01
  status: "False"
  message: update for condition
```

whereas here they are: message, status, type.

That said, maybe just not changing the ordering will fix it? I am not sure.

Reply:
hmmm... I see
right now we keep the order of fields of the conditions that already exist in the input's Kptfile, but for the new conditions added to the list by the KRM functions the order of fields are random/arbitrary

Reply:
Example of commit by *nephio-controller*:

![image](https://github.com/nephio-project/nephio/assets/9468083/c3ecfde9-13df-40d7-b441-d52f04bcc830)


Example of follow up commit "fixing" the order, by *package variant controller*:

![image](https://github.com/nephio-project/nephio/assets/9468083/130125ed-7012-40f8-9bfc-d1829add58d3)

Notice how not only the fields of the conditions re-order, but even the Kptfile `description` moves.

If this PR keeps Kptfile ordering in this sense (field-by-field), then it should help.

Reply:
where should be the "reason" field in the above order? (type, status, message)

Reply:
If we keep the existing ordering, then I think we will be ok. It may thrash once or twice, but then settle down. That is good enough for now.

Reply:
but if the problem was that KRM functions are "fighting" with Porch, where both modifies the order of fields back-and-forth: this shouldn't happen anymore. 
What should happen instead is that Porch modifies the field order of the condition created by a KRM function, but after  the KRM functions should leave the field order as-is

Reply:
> right now we keep the order of fields of the conditions that already exist in the input's Kptfile, but for the new conditions added to the list by the KRM functions the order of fields are random/arbitrary

The last comment is in response to this. If this is true, of the condition fields and the Kptfile fields (e.g, description), then we should be OK.

Reply:
that is the theory, yes ...

let's see it in practice :)

Reply:
ok!

/approve
/lgtm